src/acl/external/kerberos_sid_group/ext_kerberos_sid_group_acl.pl.in

```diff
@@ -0,0 +1,252 @@
+It uses B<ldapsearch> from Openldap to lookup the name of a AD group sid.
+
+This helper must be used in with the negotiate_kerberos_auth helper in a
+Microsft AD or Samba environement.
+
+It reads from the standard input the domain username and a list of group sids
+and tries to match the group SIDs to the AD group sids.
+
+=head1 OPTIONS
+
+=over 12
+
+=item B<-d>
+
+Write debug info to stderr.
+
+=item B<-h>
+
+Print the help.
+
+=item B<-p principal name>
+
+Principal name in squid keytab to use for ldap authentication to AD 
+
+=item B<-D domain controller>
+
+Domain controller to contact to lookup group SID
+
+=item B<-b base DN>
+
+Base DN for ldap search
+
+=item B<-G AD group name>
+
+AD group name to be used for SID lookup. List seperated by :
+
+=back
+
+=head1 CONFIGURATION
+
+  auth_param negotiate program /path/to/negotiate_wrapper_auth -d --ntlm /path/to/ntlm_auth --helper-protocol=squid-2.5-ntlmssp --domain example.com --kerberos /path/to/negotiate_kerberos_auth -d -s GSS_C_NO_NAME -k /path/to/squid.keytab -t none
+  external_acl_type sid_check ipv4 %LOGIN %note{group} /path/to/kerberos_sid_group_acl -p principal -D dc1.example.com -b "DC=example,DC=com" -G Group1:Group2
+  acl squid_allow external sid_check
+  acl allowed_group external sid_check 
+  http_access allow allowed_group
+
+If the local perl interpreter is in a unusual location it may need to be added:
+
+  external_acl_type sid_check ipv4 %LOGIN %note{group} /path/to/perl /path/to/kerberos_sid_group_acl -p principal -D dc1.example.com -b "DC=example,DC=com" -G Group1:Group2
+
+=head1 AUTHOR
+
+This program was written by Markus Moeller <markus_moeller@compuserve.com>
+
+This manual was written by Markus Moeller <markus_moeller@compuserve.com>
+
+=head1 COPYRIGHT
+
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+
+ This program is put in the public domain by Markus Moeller
+ <markus_moeller@compuserve.com>. It is distributed in the hope that it will
+ be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+=head1 QUESTIONS
+
+Questions on the usage of this program can be sent to the I<Squid Users mailing list <    squid-users@lists.squid-cache.org>>
+
+=head1 REPORTING BUGS
+
+Bug reports need to be made in English.
+See http://wiki.squid-cache.org/SquidFaq/BugReporting for details of what you need to     include with your bug report.
+
+Report bugs or bug fixes using http://bugs.squid-cache.org/
+
+Report serious security bugs to I<Squid Bugs <squid-bugs@lists.squid-cache.org>>
+
+Report ideas for new improvements to the I<Squid Developers mailing list <squid-dev@li    sts.squid-cache.org>>
+
+=head1 SEE ALSO
+
```

Comment:
Since this helper is described earlier as requiring use alongside the negotiate_kerberos_auth helper, there should probably be a reference from here to:  negotiate_kerberos_auth(8)

Reply:
Thank you 
Markus